### ISSUE TITLE
Memory stat seems to be off running in MacOS with docker agent

### ISSUE THREAD

**Issue by austinliuu** [henrygd/beszel#287]
I was running the binary which was reporting the correct RAM usage but once I switched to docker agent, the RAM usage seems only reports what docker container/services are using. I'm running MacOS 15.1 with Orbstack+Portainer

<img width="1417" alt="image" src="https://github.com/user-attachments/assets/582e85bc-bb4a-4174-8406-e12a8e11bb09">

**Comment by henrygd** [comment]
Thanks for reporting this.

Mac and Linux have different methods for calculating memory usage. When you run the binary, it uses the native method for Mac.

When you run on Docker / Orbstack on Mac, I'm assuming it uses the Linux method (directly reading `/proc/meminfo`), and the values are incorrect due to how `/proc` is implemented. This may even differ between Docker Desktop, Orbstack, and Podman.

From a quick google, Orbstack says it has its own "dynamic memory management" implementation, so it may have something to do with that.

Unfortunately I can't troubleshoot because I don't have a MacOS device. It would be helpful to confirm the `meminfo` values, though. `beszel-agent` doesn't have a shell, but if you have another orbstack container running you should be able to do this:

```bash
docker exec -it container-name sh
```

```bash
cat /proc/meminfo
```

**Comment by austinliuu** [comment]
I have run the `meminfo` commend inside the Portainer container (which I used to deploy beszel-agent), here is the output

```
❯ cat /proc/meminfo
MemTotal:        5441096 kB
MemFree:         5003708 kB
MemAvailable:    5025512 kB
Buffers:             392 kB
Cached:           164196 kB
SwapCached:            0 kB
Active:            78824 kB
Inactive:         189504 kB
Active(anon):      10672 kB
Inactive(anon):   113888 kB
Active(file):      68152 kB
Inactive(file):    75616 kB
Unevictable:           0 kB
Mlocked:               0 kB
SwapTotal:       6489664 kB
SwapFree:        6489664 kB
Dirty:               160 kB
Writeback:             0 kB
AnonPages:        101764 kB
Mapped:            67308 kB
Shmem:             20820 kB
KReclaimable:      30512 kB
Slab:              73840 kB
SReclaimable:      30512 kB
SUnreclaim:        43328 kB
KernelStack:        5408 kB
PageTables:         2392 kB
SecPageTables:         0 kB
NFS_Unstable:          0 kB
Bounce:                0 kB
WritebackTmp:          0 kB
CommitLimit:     9210212 kB
Committed_AS:     758632 kB
VmallocTotal:   135288315904 kB
VmallocUsed:       26620 kB
VmallocChunk:          0 kB
Percpu:             2520 kB
AnonHugePages:         0 kB
ShmemHugePages:        0 kB
ShmemPmdMapped:        0 kB
FileHugePages:         0 kB
FilePmdMapped:         0 kB
```

**Comment by austinliuu** [comment]
From the output the number seems to match with was I see in MacOS activity monitor:
<img width="266" alt="image" src="https://github.com/user-attachments/assets/c01a40bf-0bf9-4341-b94f-613199cab3ea">

**Comment by austinliuu** [comment]
I'm not sure if these number match with what MacOS shows, here is the number from the activity manager
<img width="273" alt="image" src="https://github.com/user-attachments/assets/f85c3b18-b316-4e81-8ff0-2732c8cb99e3">

**Comment by austinliuu** [comment]
Yeah that makes sense, but I’m having trouble setting the binary to auto start on boot up in MacOS(it needs to be done by using .plist) and my binary agent seems to stop sending telematics after some time sitting idle. Is there a way to debug why? I don’t see any error when running in terminal

**Comment by henrygd** [comment]
You're not the only one who ran into that. There's another issue here: #248

I don't know why that would happen. Maybe it needs to continuously print output or interact with the filesystem for it to be considered active?

Try adding `LOG_LEVEL=debug` and see if it continues to run when it's logging things. Otherwise let me know what it logs just before it terminates.

**Comment by svenvg93** [comment]
No response from the OP for a while — closing this issue. If the problem persists, please open a new one.
